woocommerce/woocommerce-android
# Update mocks for `My Store`, `Orders` and `Reviews` screens.

This is a continuation of mocks update process, started in #4966.

### Description
In similarity to login-related mocks, a big number of mocks used to generate data for `My Store`, `Orders` and `Reviews` screens was outdated or did not exist. In some cases, from what I saw, the data previously presented by a single response is now presented by several requests/responses.

Some mock files seem to be no longer in use, but I'd prefer to delete them in a separate PR.

**I'm sorry for the huge number of affected files, but it made sense for me to deliver a testable result.** _I'd suggest that the best way to test the changes in mocks is to run the test and explore the app_ (see below).

### A note on logical sense of displayed data.
The main goal was to create a data to show on screens. In some cases it makes no logical sense: the amount of visits per day is larger than amount of visits per month, the names of customers are random letter sometimes and have no picture for avatar, there are orders from future (see more details below).

As tests will be developed, the data can be changed easily. For now this is a placeholder.

### Testing instructions

1. Make sure you use the latest revision of mobile secrets
2. Use the `gradle.properties` from mobile secrets
3. To avoid possible Emulator flakiness, create a fresh Emulator instance. I tested under `Pixel 4a API 30`
4. Run the `mocksDemo` test that is temporary present in `ScreenshotTest.kt`. It's there only as a mean to have the app running with mocks active. Since WireMock is not running as a standalone process in this project (like in WPiOS, for example), the mocks will be in use only when the test that is an extension of `TestBase` is running.
5. Once the test logs into the app and makes a 1000 seconds pause, you can explore the app:

✅ `My Store` screen is showing Stats, Visits, Orders and Top Performers for Day / Week / Month / Year
✅ Every individual column in stats can be clicked, and data for this individual column will be shown
<img width="269" alt="Screenshot 2021-10-18 at 23 09 08" src="https://user-images.githubusercontent.com/73365754/137799770-449a357b-80d2-40e4-bd7b-52f80f5a946d.png">

✅ `Orders` screen loads the orders. The orders can be opened individually. Pictures are shown:
<img width="270" alt="Screenshot 2021-10-18 at 23 18 52" src="https://user-images.githubusercontent.com/73365754/137800889-6db47108-da98-4155-9fcb-8b2a2202bb19.png">

✅ `Reviews` screens loads the reviews. Individual review can be opened:
<img width="270" alt="Screenshot 2021-10-18 at 23 18 23" src="https://user-images.githubusercontent.com/73365754/137801037-a06a8f6f-f393-45af-8b9f-bcada013cd6a.png">

❗ ❎ ❗ Opening `Orders` screen causes a crash (only under mocks). It was so initially, and I still have to find out why. It's a topic of a separate PR.



### Note on hardcoded dates.
Whenever possible, I tried to avoid hardcoded years (and months/days when it was necessary). For example, the stats for year will always use the current year, but the months are hardcoded. Same for the dates for orders.

### Note on mocks for Week / Month stats
Since the request to show the stats for week and for month is the almost same (it selects a different interval of days), both stats for week and for month are generated by one file [reports_revenue_stats-day.json](WooCommerce/src/androidTest/assets/mocks/mappings/jetpack-blogs/wc-analytics/reports_revenue_stats-day.json), so they will look the same. The change needed to tell these two requests apart (complicated by case when the current day of the month is less than 8) would take too much time, but it can be implemented if there's a need in the future.
